lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorImplementation.cxx

```diff
@@ -100,26 +101,25 @@ void HSICEstimatorImplementation::computeCovarianceMatrices()
   outputCovarianceMatrix_ = covarianceModelCollection_[inputDimension_].discretize(outputSample_);
 }
 
-
-/* Compute the weight matrix from the weight function */
-SquareMatrix HSICEstimatorImplementation::computeWeightMatrix(const Sample&) const
+/** Compute the weights from the weight function */
+Point HSICEstimatorImplementation::computeWeights() const
 {
-  throw NotYetImplementedException(HERE) << "Use the class HSICEstimatorConditionalSensitivity in order to perform conditional sensitivity analysis";
+  return weights_;
 }
 
 /* Compute a HSIC index (one marginal) by using the underlying estimator (biased or not) */
-Scalar HSICEstimatorImplementation::computeHSICIndex(const CovarianceMatrix & covMat1,
-    const CovarianceMatrix & covMat2,
-    const SquareMatrix & weightMatrix) const
+Scalar HSICEstimatorImplementation::computeHSICIndex(const CovarianceMatrix &covMat1,
+                                                     const CovarianceMatrix &covMat2,
+                                                     const Point &weights) const
```

Comment:
here too, the computeHSICIndex docstring should be updated

Reply:
I guess it is already done, isn't it?

Reply:
It is a private method ;-)